pip/baseparser.py

```diff
@@ -132,7 +132,7 @@ def __init__(self, *args, **kwargs):
         self.name = kwargs.pop('name')
 
         isolated = kwargs.pop("isolated", False)
-        self.config = Configuration(isolated)
+        self._configuration_obj = Configuration(isolated)
```

Comment:
We already know it's an object here, by convention (and pretty much everything in python is one), so this naming is redundant.

If you really wish to indicate it's a private member then I'd just rename it `_config`.